resolwe/flow/managers/listener.py

```diff
@@ -460,7 +462,17 @@ def handle_finish(self, obj):
                     async_to_sync(self._send_reply)(obj, {ExecutorProtocol.RESULT: ExecutorProtocol.RESULT_ERROR})
                     return
 
-                if process_rc == 0 and not d.status == Data.STATUS_ERROR:
+                if spawning_failed:
+                    process_error = copy.copy(d.process_error)
+                    process_error.append("Error while preparing spawned Data objects")
+
+                    changeset = {
+                        'status': Data.STATUS_ERROR,
+                        'process_progress': 100,
+                        'finished': now(),
```

Comment:
You can set `process_progress` and `finished` before the if.

Reply:
I would also prefere to set `process_progress` in `handle_update` when status is changed to `DONE` or `ERROR`. But this is out of scope for this PR.